src/loaders/IVolumeLoader.ts

```diff
@@ -49,18 +49,21 @@ export interface IVolumeLoader {
 
   /**
    * Create an empty `Volume` from a `LoadSpec`, which must be passed to `loadVolumeData` to begin loading.
+   * Optionally pass a callback to respond whenever new channel data is loaded into the volume.
    *
-   * May cache some values for use on only the next call to `loadVolumeData`; callers should guarantee that the next
-   * call to this loader's `loadVolumeData` is made on the returned `Volume` before the volume's state is changed.
+   * Loaders are allowed to assume that they will only be called on a single data source, in order to cache
+   * information about that source. Once this method has been called, every subsequent call to it or
+   * `loadVolumeData` should reference the same source.
```

Comment:
Good to note. Interesting/scary imagining trying to enforce this in code.

Reply:
This kind of dovetails with another concern I have about this PR: `loadSpecRequired` has no reason to carry around a URL. In a world where loading from one place and then another may cause problems more often than not, it might make sense to remove URL from `LoadSpec` and instead pass it into loader constructors.

Reply:
I kinda like that idea.

Reply:
That could be a separate PR and combined with changing the subpath to be multiresolution level.